app/models/text_component.rb

```diff
@@ -9,6 +9,8 @@ class TextComponent < ActiveRecord::Base
 
   accepts_nested_attributes_for :triggers
 
+  before_create :assign_default_channel
```

Comment:
Please, don't auto-assign a channel in `before_create`. I would use `before_validation` but actually I wouldn't auto-assign a channel in this particular case at all. This will be called whenever we create a text component! It will affect all our tests. Imagine you do `create(:text_component, channel: nil)` and suddenly the component is assigned to a channel.

I suggest to use a validation `validates :channel, presence: true` (+test) and to set a default channel in the view, e.g. `f.association :channel, allow_blank: false`. (+test, either a view test or a cucumber test).